### PR TITLE
Test remote shares ocs endpoint

### PR DESCRIPTION
**Pull request by individual-it**
## Description
more tests for the  `/remote_shares` and  `/remote_shares/pending/` endponts

## Related Issue
part of #34566

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiFederation/federated.feature [hunk 1]
```diff
@@ -66,6 +66,53 @@ Feature: federated
       | mountpoint  | {{TemporaryMountPointName#/textfile0.txt}} |
       | accepted    | 0                                          |
 
+  Scenario: Remote sharee requests informations of only one share
```
Comment (phil-davis) [inline tests/acceptance/features/apiFederation/federated.feature]: s/informations/information

File: tests/acceptance/features/apiFederation/federated.feature [hunk 2]
```diff
@@ -66,6 +66,53 @@ Feature: federated
...
+      | name        | /textfile0.txt     |
+      | owner       | user0              |
+      | user        | user1              |
+      | mountpoint  | /textfile0 (2).txt |
+      | accepted    | 1                  |
+      | type        | file               |
+      | permissions | 27                 |
+
+  @issue-34636
+  Scenario: Remote sharee requests informations of only one share before accepting it
```
Comment (phil-davis) [inline tests/acceptance/features/apiFederation/federated.feature]: s/informations/information

File: tests/acceptance/features/bootstrap/FederationContext.php [hunk 3]
```diff
@@ -130,6 +133,55 @@ public function userFromServerHasAcceptedLastPendingShare($user, $server) {
 		$this->featureContext->theOCSStatusCodeShouldBe('100');
 	}
 
+	/**
+	 * @When /^user "([^"]*)" gets informations of last federated cloud share using the sharing API$/
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/FederationContext.php]: s/get informations of last federated cloud share/retrieves the information of the last federated cloud share

e.g. we have existing steps like:
```
@When /^the administrator retrieves the information of user "([^"]*)" using the provisioning API$/
```
so we can be consistent with that.

File: tests/acceptance/features/bootstrap/FederationContext.php [hunk 4]
```diff
@@ -130,6 +133,55 @@ public function userFromServerHasAcceptedLastPendingShare($user, $server) {
...
+		$this->featureContext->userSendsHTTPMethodToOcsApiEndpointWithBody(
+			$user,
+			'GET',
+			"/apps/files_sharing/api/v1/remote_shares/{$share_id}",
+			null
+		);
+	}
+
+	/**
+	 * @When /^user "([^"]*)" gets informations of last pending federated cloud share using the sharing API$/
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/FederationContext.php]: s/get informations of last pending federated cloud share/retrieves the information of the last pending federated cloud share

**Review by phil-davis — APPROVED** [review]

**Comment by individual-it** [comment]
backport in #34644

**Pull request merged**